internal/common/rules/globalrulescache_test.go

```diff
@@ -1,19 +1,89 @@
 package rules
 
 import (
+	"github.com/SOMAS2020/SOMAS2020/pkg/testutils"
+	"github.com/pkg/errors"
 	"gonum.org/v1/gonum/mat"
 	"testing"
 )
 
 // TestRegisterNewRule Tests whether the global rule cache is able to register new rules
 func TestRegisterNewRule(t *testing.T) {
-	registerTestRule()
-	if _, ok := AvailableRules["Kinda Test Rule"]; !ok {
+	AvailableRulesTesting, _ := generateTestStores()
+	registerTestRule(AvailableRulesTesting)
+	if _, ok := AvailableRulesTesting["Kinda Test Rule"]; !ok {
 		t.Errorf("Global rule register unable to register new rules")
 	}
 }
 
-func registerTestRule() {
+func TestPullRuleIntoPlay(t *testing.T) {
+	AvailableRulesTesting, RulesInPlayTesting := generateTestStores()
+	registerTestRule(AvailableRulesTesting)
+	_ = pullRuleIntoPlayInternal("Kinda Test Rule 2", AvailableRulesTesting, RulesInPlayTesting)
+	cases := []struct {
+		name string
+		rule string
+		want error
+	}{
+		{
+			name: "normal working",
+			rule: "Kinda Test Rule",
+			want: nil,
+		},
+		{
+			name: "unidentified rule name",
+			rule: "Unknown Rule",
+			want: errors.Errorf("Rule '%v' is not available in rules cache", "Unknown Rule"),
+		},
+		{
+			name: "Rule already in play",
+			rule: "Kinda Test Rule 2",
+			want: errors.Errorf("Rule '%v' is already in play", "Kinda Test Rule 2"),
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			got := pullRuleIntoPlayInternal(tc.rule, AvailableRulesTesting, RulesInPlayTesting)
+			testutils.CompareTestErrors(tc.want, got, t)
+		})
+	}
+}
+
+func TestPullRuleOutOfPlay(t *testing.T) {
+	AvailableRulesTesting, RulesInPlayTesting := generateTestStores()
+	registerTestRule(AvailableRulesTesting)
+	_ = pullRuleIntoPlayInternal("Kinda Test Rule", AvailableRulesTesting, RulesInPlayTesting)
+	cases := []struct {
+		name string
+		rule string
+		want error
+	}{
+		{
+			name: "normal working",
+			rule: "Kinda Test Rule",
+			want: nil,
+		},
+		{
+			name: "Rule already in play",
+			rule: "Kinda Test Rule 2",
+			want: errors.Errorf("Rule '%v' is not in play", "Kinda Test Rule 2"),
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			got := pullRuleOutOfPlayInternal(tc.rule, AvailableRulesTesting, RulesInPlayTesting)
+			testutils.CompareTestErrors(tc.want, got, t)
+		})
+	}
+}
+
+func generateTestStores() (map[string]RuleMatrix, map[string]RuleMatrix) {
```

Comment:
this function would be available in all test files for this package, might want to name it more specifically? 

I don't see the need for this function too tbh (you can just pass in `map[string]RuleMatrix{}` where it's needed).